components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFeature.kt

```diff
@@ -0,0 +1,77 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.qr
+
+import android.Manifest.permission.CAMERA
+import android.content.Context
+import android.support.annotation.VisibleForTesting
+import android.support.v4.app.FragmentManager
+import mozilla.components.support.ktx.android.content.isPermissionGranted
+
+typealias OnScanResult = (result: String) -> Unit
+typealias OnNeedToRequestPermissions = (permissions: Array<String>) -> Unit
+
+/**
+ * Feature implementation that provides QR scanning functionality via the [QrFragment].
+ *
+ * @property applicationContext a reference to the application context.
+ * @property fragmentManager a reference to a [FragmentManager], used to start the [QrFragment].
+ * @property onScanResult a callback invoked with the result of the QR scan. The callback will
+ * always be invoked on the main thread.
+ * @property onNeedToRequestPermissions a callback invoked when permissions
+ * need to be requested before a QR scan can be performed. Once the request
+ * is completed, [onPermissionsResult] needs to be invoked.
+ */
+class QrFeature(
+    private val applicationContext: Context,
```

Comment:
nit: Technically this doesn't need to be the application context right? Since we pass in a fragment manager this seems to have a reference to an activity context already?